atomic_reactor/core.py

```diff
@@ -231,6 +232,10 @@ def retry(function, *args, **kwargs):
                 raise
 
 
+class RetryGeneratorException(Exception):
+    """Retry generator pull/push failed"""
```

Comment:
Don't you want a constructor that takes a `reason` or `cause` parameter or something, being the original exception? e.g. https://github.com/projectatomic/osbs-client/blob/0.43/osbs/exceptions.py#L20, or https://github.com/shazow/urllib3/blob/1.22/urllib3/exceptions.py#L76

Reply:
ok will safe original error, not necessary only exception but in case cmd_result.is_failed() will safe cmd_result